src/foam/nanos/approval/ApprovableAwareDAO.js

```diff
@@ -96,7 +96,9 @@ foam.CLASS({
         ApprovalRequest request = (ApprovalRequest) req.fclone();
         request.clearId();
         request.setApprover(userId);
-        getApprovalRequestDAO().inX(x).put(request);
+        // NOTE: Explicit use of system context as ApprovalRequests can be
+       // triggered by normal user UI actions.
+        getApprovalRequestDAO().inX(getX()).put(request);
```

Comment:
Will the createdBy/Agent on approval request always be system user? if we don't allow other users eg. fraud-ops to create approval request at all, might be less valuable to capture createdBy/Agent?

Reply:
Yeah, more concerned with createdFor.  ApprovalRequest should always be programatically created/trigger, so createdBy/Agent is always system, but the createdFor should continue to be the triggering user.